core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -170,24 +169,22 @@ public BlockInStream getInStream(long blockId, InStreamOptions options,
       info = masterClientResource.get().getBlockInfo(blockId);
     }
     List<BlockLocation> locations = info.getLocations();
-    List<BlockWorkerInfo> blockWorkerInfo = Collections.EMPTY_LIST;
+    List<BlockWorkerInfo> blockWorkerInfo = getEligibleWorkers();
+    if (blockWorkerInfo.isEmpty()) {
+      throw new UnavailableException(ExceptionMessage.NO_WORKER_AVAILABLE.getMessage());
+    }
     // Initial target workers to read the block given the block locations.
     Set<WorkerNetAddress> workerPool;
     // Note that, it is possible that the blocks have been written as UFS blocks
     if (options.getStatus().isPersisted()
         || options.getStatus().getPersistenceState().equals("TO_BE_PERSISTED")) {
-      blockWorkerInfo = getEligibleWorkers();
       workerPool = blockWorkerInfo.stream().map(BlockWorkerInfo::getNetAddress).collect(toSet());
-      if (workerPool.isEmpty()) {
```

Comment:
I think this previous check was fine, right? I am just concerned with calling `getEligibleWorkers` for every code path.